src/PIL/WebPImagePlugin.py

```diff
@@ -288,7 +297,8 @@ def _save(im, fp, filename):
     xmp = im.encoderinfo.get("xmp", "")
 
     if im.mode not in _VALID_WEBP_LEGACY_MODES:
-        alpha = im.mode == 'P' and 'A' in im.im.getpalettemode()
+        alpha = 'A' in im.mode or 'a' in im.mode \
+                or im.mode == 'P' and 'A' in im.im.getpalettemode()
```

Comment:
While this is presumably what you meant to infer with the line break, could I interest you in adding brackets?
```
alpha = 'A' in im.mode or 'a' in im.mode \
         or (im.mode == 'P' and 'A' in im.im.getpalettemode())
```
Maybe it's just me, but I didn't instinctively know what `bool or bool and bool` was going to result in.